timelock-agent/src/main/java/com/palantir/atlasdb/timelock/paxos/Dependencies.java

```diff
@@ -50,6 +50,12 @@
         PaxosRemoteClients remoteClients();
         LocalPaxosComponents components();
         int quorumSize();
+
+        /**
+         * Caution! The shared executor should only be used for tasks that are expected to complete quickly.
+         * DO NOT use the shared executor when the concurrency of requests may be very high (e.g. for Paxos round
+         * verification). This may lead to thread explosion.
+         */
```

Comment:
can we slap a limit on the executor?

Reply:
like, fix it at max 256 threads or sth?

Reply:
Yeah, that sounds like a good thing to do

Reply:
Since the claim is that it makes sense to share the executors for the other endpoints as they are explicitly _not_ heavily barraged

Reply:
I'll have a look on internal metrics framework to decide what a good threshold might be (though might still need monitoring further since that will include acceptor activity which is the bulk of the activity).

Reply:
Yeah, internal metrics not super helpful - they do spike above 400+ in asymptomatic operation (and over 1000 on the relevant ticket), but I'm pretty confident a majority of these are acceptor operations.